packages/account/src/connectors/fuel.ts

```diff
@@ -101,11 +102,21 @@ export class Fuel extends FuelConnector implements FuelSdk {
     this._storage = config.storage === undefined ? this.getStorage() : config.storage;
     // Setup all methods
     this.setupMethods();
-    // Get the current connector from the storage
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this.setDefaultConnector();
-    // Setup new connector listener for global events
-    this._targetUnsubscribe = this.setupConnectorListener();
+    this._initializationPromise = this.initialize();
+  }
+
+  private async initialize(): Promise<void> {
+    try {
+      await this.setDefaultConnector();
+      this._targetUnsubscribe = this.setupConnectorListener();
+    } catch (error) {
+      throw new FuelError(ErrorCode.INVALID_PROVIDER, 'Error initializing Fuel Connector');
+    }
+  }
+
+  public async init(): Promise<Fuel> {
```

Comment:
From https://github.com/FuelLabs/fuels-ts/issues/2892#issuecomment-2294451450:
> To avoid introducing breaking the constructor for now I've added a new [init](https://github.com/FuelLabs/fuels-ts/pull/2930/files#diff-f2f37286dc75888286dbb2da3164a73b4a4a1ea8a90d87fd1ec9bcee1666097cR117-R119) method which will allow a consumer to await all initialization before continuing.

One could say that this is a breaking change because fully instantiating a connector now requires an additional step. If nothing, we should at least update the documentation accordingly as this is unintuitive.

Reply:
> One could say that this is a breaking change because fully instantiating a connector now requires an additional step.

I disagree, this change does not cause any failures for consumers nor does it **require** changes, although it is recommended. Consumers who are instantiating the constructor as is and are not experiencing race conditions will remain unaffected. 

> If nothing, we should at least update the documentation accordingly as this is unintuitive.

I agree here, I've updated the docs to recommend this method of instantiating the class.

dac5906